### DISCUSSION TITLE
velero backup status is ending in Partially Failed waiting for openstack snapshots to become available.

### DISCUSSION THREAD

**Discussion by JuanGarcia01** [vmware-tanzu/velero#8140] [Community support Q&A] [answered]
**What steps did you take and what happened:**
Upgraded Velero from version  1.11.0. to version  --> 1.14.0
Upgraded velero-plugin-for-openstack:v0.5.2 to version --> velero-plugin-for-openstack:v0.7.0

**The following information will help us better understand what's going on**:

We are seeing an issue with Velero backups are Partially Failing because they are erroring out or timing out waiting for native velero snapshots that are created in openstack to become available.  

The environment we are hosting in openstack has some very large PVs (500Gb x 1 , 150Gb x 2 , 30G x 1 ) so it can take 10+ mins for the large snapshots that are created to become.

The error message we are seeing in the backup details...
**message: /snapshot didn't get into 'available' state within the time limit**

Eventually the volume snapshots that are created  by the velero backup in openstack are getting to an available state it is just taking a very long time.

It has always taken a long time for the large snapshots to be created in openstack. 
However with the previous version the velero backups would not time out or error out waiting for the snapshot creation to complete. 

It would appear that the maybe backup timeout settings are different for the with the latest version of velero ?
How do we change the timeout settings for openstack velero backups.

current Kubernetes version 1.28.6

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "I would like to see this bug fixed as soon as possible"
- :-1: for "There are more important bugs to focus on right now"

**Comment by blackpiglet** [comment]
Are you using `velero-plugin-for-openstack` to create snapshots in the Velero-native snapshot way?

If so, that always works as you described because the `VolumeSnapshotter` never supports querying the progress of the snapshot.

**Comment by JuanGarcia01** [comment, marked as answer]
After using this as a reference the issue is now fixed
openstack backups are now completing after increasing  timeouts for
      volumeTimeout: 5m
      snapshotTimeout: 5m
      cloneTimeout: 5m
      backupTimeout: 5m
      imageTimeout: 5m
      
 **Reference**
https://github.com/Lirt/velero-plugin-for-openstack/blob/v0.7.0/docs/installation-using-helm.md
